warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/edge/EdgeEventFieldUtil.java

```diff
@@ -0,0 +1,241 @@
+            NormalizedContentInterface downnci = getNullKeyedNCI(edgeDef.getDownTime(), normalizedFields);
+            if (null != upnci && null != downnci) {
+                edgeDataBundle.initDuration(upnci, downnci);
+            }
+        } else {
+            NormalizedContentInterface elnci = getNullKeyedNCI(edgeDef.getElapsedTime(), normalizedFields);
+            if (null != elnci) {
+                edgeDataBundle.initDuration(elnci);
+            }
+        }
+    }
+
+    protected void setEdgeKeyDateType(EdgeDataBundle bundle, boolean validActivtyDate, boolean sameActivityDate, long eventDate, long newFormatStartDate) {
+        EdgeKey.DATE_TYPE dateType = null;
+        if (eventDate < newFormatStartDate) {
+            dateType = EdgeKey.DATE_TYPE.OLD_EVENT;
+        } else if (validActivtyDate) {
+            if (sameActivityDate) {
+                dateType = EdgeKey.DATE_TYPE.ACTIVITY_AND_EVENT;
+            } else {
+                dateType = EdgeKey.DATE_TYPE.ACTIVITY_ONLY;
+                // also need to write EVENT_ONLY for some reason
+            }
+        } else {
+            dateType = EdgeKey.DATE_TYPE.EVENT_ONLY;
+        }
+        bundle.setDateType(dateType);
+    }
+
+    public EdgeDataBundle setEdgeInfoFromEventFields(EdgeDataBundle bundle, EdgeDefinitionConfigurationHelper edgeDefConfigs, RawRecordContainer event,
+                    EdgeIngestConfiguration edgeConfig, long newFormatStartDate, Configuration conf, String typeName) {
+
+        // Get the load date of the event from the fields map
+        String loadDateStr = getLoadDateString(normalizedFields);
+        long activityDate = getActivityDate(normalizedFields, edgeDefConfigs);
+
+        // get the activity date from the event fields map
+        boolean validActivityDate = validateActivityDate(activityDate, event.getDate(), edgeConfig.getPastDelta(), edgeConfig.getFutureDelta());
+        boolean activityEqualsEvent = false;
+
+        // If the activity date is valid check to see if it is on the same day as the event date
+        if (validActivityDate) {
+            activityEqualsEvent = compareActivityAndEvent(activityDate, event.getDate());
+        }
+
+        if (event.getAltIds() != null && !event.getAltIds().isEmpty()) {
+            bundle.setUUID(event.getAltIds().iterator().next());
+        }
+
+        setEdgeKeyDateType(bundle, validActivityDate, activityEqualsEvent, event.getDate(), newFormatStartDate);
+
+        bundle.setLoadDate(loadDateStr);
+        bundle.setActivityDate(activityDate);
+        bundle.setEventDate(event.getDate());
+        bundle.setValidActivityDate(validActivityDate);
+        bundle.setEdgeAttribute2(getEdgeAttr2(edgeDefConfigs));
+        bundle.setEdgeAttribute3(getEdgeAttr3(edgeDefConfigs));
+        bundle.setRequiresMasking(event.isRequiresMasking());
+        bundle.setHelper(event.getDataType().getIngestHelper(conf));
```

Comment:
Looking for opportunities to drop more method parameters - it looks like the `conf` parameter is only used here.  I suggest updating EdgeIngestConfiguration to save its constructor argument (which is the same as this conf, retrieved from context.getConfiguration()), so that it's only necessary to use EdgeIngestConfiguration.